src/NServiceBus.Core/Routing/RoutingSettings.cs

```diff
@@ -25,6 +27,38 @@ public void RouteToEndpoint(Type messageType, string destination)
         {
             Settings.GetOrCreate<UnicastRoutingTable>().RouteToEndpoint(messageType, destination);
         }
+
+        /// <summary>
+        /// Adds a static unicast route for all types contained in the specified assembly.
+        /// </summary>
+        /// <param name="assembly">The assembly whose messages should be routed.</param>
+        /// <param name="destination">Destination endpoint.</param>
+        public void RouteToEndpoint(Assembly assembly, string destination)
+        {
+            var routingTable = Settings.GetOrCreate<UnicastRoutingTable>();
+            foreach (var type in assembly.GetTypes())
+            {
+                routingTable.RouteToEndpoint(type, destination);
+            }
+        }
+
+        /// <summary>
+        /// Adds a static unicast route for all types contained in the specified assembly within the given namespace.
+        /// </summary>
+        /// <param name="assembly">The assembly whose messages should be routed.</param>
+        /// <param name="messageNamespace">The namespace of the messages which should be routed.</param>
+        /// <param name="destination">Destination endpoint.</param>
+        public void RouteToEndpoint(Assembly assembly, string messageNamespace, string destination)
+        {
+            // empty namespace is null, not string.empty
+            messageNamespace = messageNamespace == string.Empty ? null : messageNamespace;
+
+            var routingTable = Settings.GetOrCreate<UnicastRoutingTable>();
+            foreach (var type in assembly.GetTypes().Where(t => t.Namespace == messageNamespace))
```

Comment:
one thing to note here is that we cannot rely at this point on the conventions to filter for message types, so we either skip that filter or delay that until the feature setup when we're sure the conventions have been configured. Thoughts?